File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/adverts/UpdateAdvertCommand.java
package by.fpmibsu.findafriend.controller.commands.adverts;

import by.fpmibsu.findafriend.application.mediatr.Request;
import by.fpmibsu.findafriend.entity.Advert;
import by.fpmibsu.findafriend.entity.Place;

import java.util.Date;

public class UpdateAdvertCommand extends Request<Boolean> {
    public int advertId;
    public Integer advertType;
    public String title;
    public String description;
    public Date creationDate;
    public Place place;

    public UpdateAdvertCommand(int advertId, Integer advertType, String title, String description, Date creationDate, Place place) {
        this.advertId = advertId;
        this.advertType = advertType;
        this.title = title;
        this.description = description;
        this.creationDate = creationDate;
        this.place = place;
    }
}

File: BackendNET/Backend/Commands/Adverts/UpdateAdvert/UpdateAdvertCommand.cs
using Backend.Entities;
using MediatR;

namespace Backend.Commands.Adverts.UpdateAdvert;

public class UpdateAdvertCommand : IRequest<bool>
{
    public int AdvertId { get; set; }
    public Advert.AdvertType? AdvertType { get; set; }
    public string? Title { get; set; }
    public string? Description { get; set; }
    public int? PlaceId { get; set; }
}